R/loadFromMetaphlan.R

```diff
@@ -149,20 +149,21 @@ loadFromMetaphlan <- function(file, sample_meta = NULL, phy_tree = NULL, ...){
 }
 
 # Check that metaphlan file contains correct information
-.check_metaphlan <- function(data){
-    # Get rowdata columns
-    rowdata_columns <- data[ , 1:2]
-    # Get columns that go to assay
-    assay_columns <- data[ , 3:ncol(data)]
+.check_metaphlan <- function(x){
+    # retrieving field types present in the data 
+    col_types <- sapply(x, class)
+    if(any(!(col_types %in% c("character", "numeric")))) {
+      warning("File contains other than character and numeric fileds.")
+      }
     # Initialize result 
     result <- TRUE
     
-    # Check rowdata column names that they contain right information, and check that 
-    # rest of the columns represents abundances in samples.
+    # Check presence of character fields to be used for rowData
+    # and similarly numeric fields for abundance assay data.  
+    # Check rowData column names that they contain right information.
     # If these requirements are met, give FALSE. Otherwise, give TRUE.
-    if( any(colnames(rowdata_columns) %in% "clade_name") && 
-        any(grepl("id", colnames(rowdata_columns))) && 
-        is.numeric(unlist(assay_columns)) ){
+    if( all(c("character", "numeric") %in% col_types) &&
```

Comment:
do we need this check any more because it has already been checked on line 155?

Reply:
yes, the first check focuses on other classes than numeric and character, there could be a chance that these two needed are not present after the first check.

Reply:
We should avoid unnecessary duplicate checks to keep the code short, clean, and more easily maintainable. If this check is not needed then it should be removed.

Reply:
I guess this checks if both rowData and assay is provided; it is ok

First one just checks that there are no factor etc values